lib/datasource/maven/index.js

```diff
@@ -132,8 +133,25 @@ async function downloadFileProtocol(pkgUrl) {
 
 async function downloadHttpProtocol(pkgUrl) {
   let raw;
+  const gotParams = [pkgUrl];
+  const hostRule = hostRules.find({
+    platform: 'maven',
+    endpoint: pkgUrl,
+  });
+  if (hostRule && hostRule.username && hostRule.password) {
+    const auth = Buffer.from(
+      `${hostRule.username}:${hostRule.password}`
+    ).toString('base64');
+    gotParams.push({
+      json: true,
```

Comment:
It looks strange to me that `json: true` is added only when Authentication is used. Should probably be never or always.

Reply:
I couldn't find `json: true` related requirements in the  [got docks](https://github.com/sindresorhus/got#json-mode).
But [gh-got readme](https://github.com/sindresorhus/gh-got#usage) in the section `you can do instead of` suggested that `json: true` is required if you are not using `gh-got`. What I tryed:
```javascript
const got = require('got');
const token = "mysecrettoken";

(async () => {
    const { body } = await got('https://api.github.com/users/Dmitry9test', {
        json: true,
        headers: {
            'accept': 'application/vnd.github.v3+json',
            'authorization': `token ${token}`
        }
    });

    console.log(body.login);
    //=> 'Dmitry9test'
})();
```
If i remove `json: true` then console outputs `undefined` instead of correct login. So I think `json: true` in this case is mandatory.

Reply:
You need json: true if you are requesting JSON from the server. A query to github's API is indeed requesting JSON but this PR is for the Maven datasource and not GitHub's API. If the Maven datasource didn't request JSON before, why why would it need it once you add Authentication? What type of data is being returned?

Reply:
`downloadMavenXml()` in this file is a pretty big hint that you're not requesting JSON

Reply:
I guess you are right. I miss looked that it is XML. I will remove `json: true`. Have you a test-repo with related dependencies configured via renovate.json to test against?